docs/KernelFeatures/UsingSecureVault.md

```diff
@@ -0,0 +1,108 @@
+# Secure Vault
+> For the full list of capabilities available in this kernel version, see the **features** section in the [root README.md file](../../README.md#key-features-and-tools). 
+
+WSO2 Carbon is shipped with a Secure Vault implementation. This allows you to store encrypted passwords that are mapped to aliases, i.e., you can use the aliases instead of the actual passwords in your configuration files for better security. For example, some configurations require the admin username and password. If the admin user password is "admin", you could use the alias `UserManager.AdminUser.Password` in your configuration file. You would then map that alias to the actual password "admin". At runtime, the product will look up this alias in the secure vault and SecureVault will return the decrypted password.
+
+There are three files that are needed by the SecureVault:
+
+1. Secure-vault.yaml: 
+    Configurations that are needed to configure the SecureVault is given in this file
+2. Master-keys.yaml: 
+    The default SecureVault implementation is based on the Java Key Store (JKS). Passwords that are needed to access the JKS and Keys are specified in this file. 
+3. Secrets.properties: 
+    This file contains the alias with plain text or encrypted password.
+    Example:
+    
+        UserManager.AdminUser.Password=plainText ABC@123
+        UserManager.AdminUser.Password=cipherText SnBSWKjtZZOo0UsmOpPRhP6ZMNYTb80+BZHRDC/kxNT9ExcTswAbFjb/aip2KgQNaVuIT27UtrBaIv77Mb5sNPGiwyPrfajLNhSOlke2p8YmMkegx/mG2ytJhJa5j9iMGtCsbMt+SAf85v6kGIiH0gZA20qDZ9jnveT7/Ifz7v0\=
+
+The SecureVatult reads the aliases and passwords given in the secrets.properties file and return the resolved (decrypted) password when it is queried wish a valid alias of a password.
+
+The SecureVault implementation has two major sub-components in it, namely Master Key Reader and Secret Repository. The SecureVault implementation allows to plugin custom implementations for both of these:
```

Comment:
The SecureVault implementation has two major sub-components, namely the Master Key Reader and Secret Repository. The SecureVault implementation allows you to plugin custom implementations for both these sub-components.